### PR TITLE
python3Packages.dugong: 3.7.5 -> 3.8.1

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 3.8.1

Change log: https://github.com/python-dugong/python-dugong/blob/master/Changes.rst#release-381-2021-01-10

Enable tests and add meta.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 129036` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python38Packages.dugong</li>
    <li>python39Packages.dugong</li>
    <li>s3ql</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129036` at d125252f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>python38Packages.dugong</li>
    <li>python39Packages.dugong</li>
    <li>s3ql</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/dugong/default.nix:30:5:
```
   |
30 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/dugong/default.nix [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
+  pythonImportsCheck = [ "dugong" ];
 
-  src = fetchPypi {
-    inherit pname version;
-    extension = "tar.bz2";
-    sha256 = "10vjdp21m0gzm096lgl84z184s5l9iz69ppj6acgsc125037dl6h";
+  meta = with lib; {
+    description = "HTTP 1.1 client designed for REST-ful APIs";
+    homepage = "https://github.com/python-dugong/python-dugong/";
+    license = licenses.mit;
```
Comment (risicle) [inline pkgs/development/python-modules/dugong/default.nix:29]: The github repo suggests it's PSF & Apache license: https://github.com/python-dugong/python-dugong/blob/74c7a078ab50d68c497797610f8c6d3644854266/LICENSE

**Comment by risicle** [comment]
Otherwise great improvement.

`nixpkgs-review` happy macos 10.15 & nixos x86_64.

File: pkgs/development/python-modules/dugong/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/dugong/default.nix:29]: Thanks a lot for checking. Looks like that I didn't check properly after coping the content.

**Pull request merged**